src/Symfony/Component/Clock/MockClock.php

```diff
@@ -62,7 +58,7 @@ public function sleep(float|int $seconds): void
         $now = substr_replace(sprintf('@%07.0F', $now), '.', -6, 0);
         $timezone = $this->now->getTimezone();
 
-        $this->now = (new \DateTimeImmutable($now, $timezone))->setTimezone($timezone);
+        $this->now = Moment::createFromInterface(new \DateTimeImmutable($now, $timezone))->setTimezone($timezone);
```

Comment:
What's the reason you're avoiding to call the constructor of `Moment` directly?

Reply:
Infinite loop : new Moment() gets the current clock, which can be a MockClock.